tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -8,18 +8,20 @@ SPDX-License-Identifier: GPL-2.0
 Copyright (C) Aqua Security inc.
 */
 
-#ifdef asm_inline
-#undef asm_inline
-#define asm_inline asm
-#endif
-
 #ifndef CORE
-
 /* In Linux 5.4 asm_inline was introduced, but it's not supported by clang.
  * Redefine it to just asm to enable successful compilation.
  * see https://github.com/iovisor/bcc/commit/2d1497cde1cc9835f759a707b42dea83bee378b8 for more details
+ * Note: types.h should be included before defining asm_inline or compilation might break
  */
 #include <linux/types.h>
+#endif
+#ifdef asm_inline
+#undef asm_inline
+#define asm_inline asm
+#endif
```

Comment:
Do we need these lines if compiling for CO:RE? I removed them and compiled with CO:RE and it doesn't seem to cause any issues.

Reply:
Not sure that we do.
Better to check that on an environment where this problem occurs (such as @mtcherni95 's) we can compile CORE succesfully when removing them.

Reply:
Ok, we tested on @mtcherni95 environment, and this is not required.
I updated the PR accordingly